src/components/KLabel/KLabel.vue

```diff
@@ -75,23 +66,32 @@ const hasTooltip = computed((): boolean => !!(props.info || props.help || slots.
   line-height: var(--kui-line-height-30, $kui-line-height-30);
   margin-bottom: var(--kui-space-40, $kui-space-40);
 
-  .is-required {
-    color: var(--kui-color-text, $kui-color-text);
-    font-size: var(--kui-font-size-30, $kui-font-size-30);
-    font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);
-    margin-left: var(--kui-space-20, $kui-space-20);
+  &.is-required {
+    margin-left: calc($kLabelSpacingX + 6px); // 6px to compensate for the 6px width of the dot
+    position: relative;
+
+    &::before {
+      background-color: var(--kui-color-background-danger, $kui-color-background-danger);
+      border-radius: var(--kui-border-radius-circle, $kui-border-radius-circle);
+      bottom: calc(50% - 2px); // place the dot in the middle of the text
+      content: '';
+      height: 6px;
+      left: 0px;
+      margin-left: calc((-1 * $kLabelSpacingX) - 6px); // -6px to compensate for the 6px width
+      position: absolute;
+      width: 6px;
+    }
   }
 
   .label-tooltip {
-    align-items: center;
-    display: flex;
+    margin-left: $kLabelSpacingX;
 
-    :deep(.kong-icon) {
-      &.kong-icon-help,
-      &.kong-icon-info {
-        cursor: pointer;
-        height: 16px;
-      }
+    .tooltip-trigger-icon {
+      cursor: pointer;
+      /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+      height: var(--kui-icon-size-30, $kui-icon-size-30) !important;
+      /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
```

Comment:
suggestion: when you're disabling the stylelint rule(s) can you please leave an additional comment as to why

Reply:
Will do. But also I'm wondering if we should allow `icon-size` tokens in `width` and `height` properties in the stylelint rule?

Reply:
I think that might be ok -- we just don't have to advertise that it's allowed

Reply:
Okay, will put up a PR with update but keeping the ignore comment for now